### PR TITLE
Changing instances of cluster name in PDO for cluster ID in fedramp e…

### PR DESCRIPTION
**Pull request by luis-falcon**
Changing instances of cluster name in PDO for cluster ID in fedramp environments

### CODE REVIEW

**Review by jewzaam — CHANGES_REQUESTED** [review]
Why not simply use the actual ID instead of name for all environments?  What is the value of having it be different?  The property set in the PD data is "ClusterID" so I'm surprised we use name.  Name is not unique and we shouldn't care about it.  The thing we care about is ID...

**Comment by sam-nguyen7** [comment]
> Why not simply use the actual ID instead of name for all environments? What is the value of having it be different? The property set in the PD data is "ClusterID" so I'm surprised we use name. Name is not unique and we shouldn't care about it. The thing we care about is ID...

@jewzaam I don't have a clear answer on this but I know that this was briefly discussed at the leads level and it was decided for only fedramp. cc @blrm 
https://coreos.slack.com/archives/C01M1NRKUF4/p1646142273310629?thread_ts=1646081822.790079&cid=C01M1NRKUF4

**Comment by jewzaam** [comment]
> > Why not simply use the actual ID instead of name for all environments? What is the value of having it be different? The property set in the PD data is "ClusterID" so I'm surprised we use name. Name is not unique and we shouldn't care about it. The thing we care about is ID...
> 
> @jewzaam I don't have a clear answer on this but I know that this was briefly discussed at the leads level and it was decided for only fedramp. cc @blrm https://coreos.slack.com/archives/C01M1NRKUF4/p1646142273310629?thread_ts=1646081822.790079&cid=C01M1NRKUF4

Understood the ask is for fedramp only.  But there is a misalignment in the parameter to PD and the value passed.  And we should not care about the cluster name.  I don't see value in having this logic specific to fedramp unless someone can state why cluster name needs to remain for non-fedramp.

@blrm @luis-falcon thoughts?

**Comment by fahlmant** [comment]
Just curious. what does the resulting PD service look like? Do you have a screenshot or something to share?

**Comment by luis-falcon** [comment]
@jewzaam  The first changes I made had non of this logic. Just a simple replacement using the clusterID instead of the name for every environment. I am of the same opinion as you that just keeping the name for non fedramp and id for fedramp is just extra steps for nothing of real impact really. I  also do not see the need for the cluster name if we had the ID or to make a distinction  but I had discuss this with blair previously but apparently in lead talks this separation was decided so i just coded it this way. Would be nice to have the reasoning though :)

**Comment by blrm** [comment]
I do agree that we should ultimately just move to cluster ID for everything. The reasoning for keeping this change scoped to only fedramp for now is explained in the JIRA card here: https://issues.redhat.com/browse/OSD-9561?focusedCommentId=19589360&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-19589360

Given the limited time we have to deliver the fedramp environment and the blast radius of swapping out all existing PD services for their Cluster ID replacements, we decided to just target fedramp initially and then move to cluster IDs for the whole fleet in a subsequent epic.

File: pkg/controller/pagerdutyintegration/clusterdeployment_created.go [hunk 1]
```diff
@@ -73,6 +73,9 @@ func (r *ReconcilePagerDutyIntegration) handleCreate(pdclient pd.Client, pdi *pa
 	}
 
 	ClusterID := cd.Spec.ClusterName
```
Comment (sam-nguyen7) [inline pkg/controller/pagerdutyintegration/clusterdeployment_created.go:75]:
```suggestion
	clusterID := cd.Spec.ClusterName
```

This variable shouldn't be exported.

File: pkg/controller/pagerdutyintegration/clusterdeployment_deleted.go [hunk 2]
```diff
@@ -58,6 +58,9 @@ func (r *ReconcilePagerDutyIntegration) handleDelete(pdclient pd.Client, pdi *pa
 	}
 
 	ClusterID := cd.Spec.ClusterName
```
Comment (sam-nguyen7) [inline pkg/controller/pagerdutyintegration/clusterdeployment_deleted.go:60]:
```suggestion
	clusterID := cd.Spec.ClusterName
```
This variable shouldn't be exported.

**Review by sam-nguyen7 — CHANGES_REQUESTED** [review]

**Comment by jewzaam** [comment]
Following up in JIRA, thanks for the link and replies.

File: pkg/controller/pagerdutyintegration/clusterdeployment_created.go (hunk 1 above)
Comment (luis-falcon) [reply pkg/controller/pagerdutyintegration/clusterdeployment_created.go:75]: good catch sam. thanks

**Review by sam-nguyen7 — COMMENTED** [review]
/lgtm
@luis-falcon any troubles with testing this before we merge?

**Review by sam-nguyen7 — APPROVED** [review]

**Comment by blrm** [comment]
@luis-falcon  If testing for this one went well, this looks good to me. @jewzaam Could you PTAL? Thanks!

**Pull request merged**